File: 2019/python/day01.py
def solve(input):
  a = [int(x) for x in input.strip().split("\n")]
  
  p1, p2 = 0, 0
  for x in a:
    p1 += x // 3 - 2
    y = x
    while y > 0:
      y = y // 3 - 2
      p2 += max(0, y)

  print("Part 1:", p1)
  print("Part 2:", p2)

# solve("100756")

import sys
locs = [x[x.find('=')+1:] for x in sys.argv if x.startswith("--input-location")] \
  + ['../inputs/day01input.txt']
solve(open(locs[0]).read())

File: 2017/rust/day01.rs
use std::fs;

fn solve(input: String) {
  let a: Vec<char> = input.trim().chars().collect();
  let n = a.len();
  let mut p1 = 0;
  let mut p2 = 0;
  for i in 0..n {
    let v = (a[i] as u32) - ('0' as u32);
    if a[i] == a[(i + 1) % n] {
      p1 += v;
    }
    if a[i] == a[(i + n / 2) % n] {
      p2 += v;
    }
  }
  println!("Part 1: {}", p1);
  println!("Part 2: {}", p2);
}

fn main() {
  // let t1 = String::from("1122");
  // solve(t1);

  // let t1 = String::from("1111");
  // solve(t1);

  let input = fs::read_to_string("../inputs/day01input.txt").unwrap();
  solve(input)
}